test/test_jit.py

```diff
@@ -7033,6 +7033,39 @@ def func():
 
         self.checkScript(func, (), optimize=True)
 
+    def test_tensor_subscript_augassign(self):
+        def bitwise_or(x):
+            x[0] |= 1
+            return x
+
+        def bitwise_and(x):
+            x[0] &= 0
+            return x
+
+        def bitwise_xor(x):
+            x[0] ^= 1
+            return x
+
+        def lshift(x):
+            x[0] <<= 1
+            return x
+
+        def rshift(x):
+            x[0] >>= 1
+            return x
+
+        def checkScriptCloneInput(func, inp):
+            cu = torch.jit.CompilationUnit(inspect.getsource(func))
+            script_fn = getattr(cu, func.__name__)
+            input_clone = inp.clone()
+            self.assertEqual(func(inp), script_fn(input_clone))
+
+        checkScriptCloneInput(bitwise_or, torch.tensor([0]))
```

Comment:
Why not just use self.checkScript?

Reply:
Because it's inplace operator, and the bug is "scripted inplace operator doing nothing".

self.checkScript reuse input for `python_fn` and `scripted_fn`, it won't tell which fn changed the input.

so I clone the input

Reply:
Ah I see that makes sense. Can you make it a utility function just like checkScript so that other people can use it in the future?

Reply:
add utility function `checkScriptCloneInput` to `class JitTestCase`